### ISSUE TITLE
What media type needs a BCP-006-xxx next?

### ISSUE THREAD

**Issue by garethsb** [AMWA-TV/nmos-stream-mappings#1]
NMOS supports compressed video streams; at the most basic via the `media_type` attribute in the Flow and the `media_types` capability in the Receiver.

Since IS-04 v1.3, we can [register new Flow attributes](https://specs.amwa.tv/nmos-parameter-registers/branches/main/flow-attributes/), and with BCP-004-01 Receiver Capabilities and the [Capabilities register](https://specs.amwa.tv/nmos-parameter-registers/branches/main/capabilities/), new Receiver capabilities. For example, for JPEG XS, it was agreed that the compression level is an important characteristic to add (as well as being able to indicate color (sub-)sampling for coded video Flows via the same `components` attribute used for raw video Flows).

The BCP-006-xxx series will include a small best practice document for each media type, to precisely define the mapping of the format into IS-04 to ensure interoperability between Senders, Receivers and Controllers.

These documents should refer to the relevant IANA media type registration, the IETF RFC for the payload format and/or SMPTE specification. The documents should only define the representation of the format parameters in IS-04 sufficient to enable a Controller to make the decision about whether a Receiver will be capable of handling a stream from a particular Sender.

AMWA members have shown interest in the following formats initially:

* [`video/jxsv`](https://www.iana.org/assignments/media-types/video/jxsv) (work in progress at [BCP-006-01](https://github.com/AMWA-TV/bcp-006-01))
* [`video/H264`](https://www.iana.org/assignments/media-types/video/H264) (see #2)
* [`video/H265`](https://www.iana.org/assignments/media-types/video/H265) (see #3)
* [`video/VP9`](https://www.iana.org/assignments/media-types/video/VP9)
* [`video/AV1`](https://www.iana.org/assignments/media-types/video/AV1)
* `audio/AM824` (which doesn't appear to have an IANA registration...)

Please comment on this issue to indicate interest in one or more of these or other formats.

Before starting a BCP document (which will each be in a new repo, like [BCP-006-01](https://github.com/AMWA-TV/bcp-006-01) for JPEG XS), an issue in this tracker can be used to discuss the necessary format parameters, etc.

**Comment by rbgodwin-nt** [comment]
Can we add video/htj2k for HTJ2K?  (reference for review for applicability to BCP-006 (http://ds.jpeg.org/whitepapers/jpeg-htj2k-whitepaper.pdf)

**Comment by garethsb** [comment]
> Can we add video/htj2k for HTJ2K?

Sure. No IANA media type registration or IETF RFC for an RTP payload mapping?

**Comment by peterbrightwell** [comment]
HTJ2K is JPEG 2000 part 15 and based on info presented by its creator in 2020, I understand it simply uses RFC 5371 RTP Payload Format for JPEG 2000 Video Streams and uses `video/jpeg2000`

**Comment by pedro-alves-ferreira** [comment]
But would a J2K-1 decoder understand it? If not, using the same media type would be a can of worms.

**Comment by peterbrightwell** [comment]
I suspect we may be into new territory here as HTJ2K projects have so far been more file-based (MXF, IMX etc) so the container maybe will identify that it's part 15.   And there is a separate JPH file format too.

Some relevant open source projects:

https://github.com/osamu620/MatHTJ2K
https://github.com/aous72/OpenJPH
https://github.com/chafey/openjphjs

**Comment by garethsb** [comment]
Adding to the list to discuss how MPEG TS should be described in NMOS, since this container may have e.g. H.264, H.265, etc. in it and may multiplex several elementary streams.

**Comment by daniel-pebble** [comment]
As I understand it the desire of this initiative is to add support for more media types to NMOS IS-04.  I believe the definition of how MPEG-TS streams map into NMOS is out of scope, however, if we have support for the common video and audio essence media types then it would go a long way to enable a mapping in future.

### Video Formats
- [video/h264](https://www.iana.org/assignments/media-types/video/H264) - Advanced Video Coding (MPEG-4 part 10)
- [video/h265](https://www.iana.org/assignments/media-types/video/H265) - High Efficiency Video Coding (MPEG-H Part 2)
- [video/mpv]( https://www.iana.org/assignments/media-types/video/MPV) - MPEG 2 Video

### Audio Formats
- [audio/mpeg](https://www.iana.org/assignments/media-types/audio/mpeg) - MPEG-1 Audio Layer II
- [audio/aac](https://www.iana.org/assignments/media-types/audio/aac) - Advanced Audio Coding (AAC)
- [audio/ac3](https://www.iana.org/assignments/media-types/audio/ac3) - Dolby D / AC3
- audio/AM824 - Dolby E or PCM

### Subtitle Formats
There doesn't seem to be many media types defined for standard broadcast subtitling formats.  
I'm assuming this one maps to DVB bitmap subs although the references spec is for DVB rather than just subtitling. 
- [image/vnd.dvb.subtitle](https://www.iana.org/assignments/media-types/text/vnd.dvb.subtitle) - DVB Bitmap

In future, it would be good to add support for DVB teletext, SCTE and any other MPEG2-TS private stream.

**Comment by garethsb** [comment]
As I understand it, there is an RTP payload mapping for MPEG TS ([`video/MP2T`](https://www.iana.org/assignments/media-types/video/MP2T)) as well as the mapping of MPEG TS directly on UDP. The RTP payload format could be in scope, let's discuss.

**Comment by daniel-pebble** [comment]
If we are going to map MPEG2 TS we'll need a way to "select" programs within TS so that a receiver could decode the essence streams for a single program in a multiple program stream.  Another option would be to just map programs using [`video/MP2P`](https://www.iana.org/assignments/media-types/video/MP2P), this would enable switching MPEG-TS "programs" which is really what we require.

**Comment by garethsb** [comment]
This should almost certainly take advantage of the existing `mux` format type in IS-04 and the ability to describe Flow `parents`.

**Comment by garethsb** [comment]
Another multiplexed format to consider whether it needs a BCP-006-xx...

* `video/SMPTE2022-6` for [ST 2022-6](https://ieeexplore.ieee.org/document/7289943) (which is another one that doesn't actually have an IANA media type registration)

**Comment by peterbrightwell** [comment]
Architecture Review Group review: Closing as there is now BCP-006-02/03 in progress and others being considered. This issue was a snapshot.
